crates/uv/src/lib.rs

```diff
@@ -1817,45 +1817,44 @@ where
         }
     };
 
-    // Windows has a default stack size of 1MB, which is lower than the linux and mac default.
+    // Running out of stack has been an issue for us. We box types and futures in various places
+    // to mitigate this, with this being an especially important case.
+    //
+    // Non-main threads should all have 2MB, as Rust forces platform consistency there,
+    // but that can be overridden with the RUST_MIN_STACK environment variable if you need more.
+    //
+    // Main thread stack-size is the real issue. There's BIG variety here across platforms
+    // and it's harder to control (which is why Rust doesn't by default). Notably
+    // on macOS and Linux you will typically get 8MB main thread, while on Windows you will
+    // typically get 1MB, which is *tiny*:
     // https://learn.microsoft.com/en-us/cpp/build/reference/stack-stack-allocations?view=msvc-170
-    // We support increasing the stack size to avoid stack overflows in debug mode on Windows. In
-    // addition, we box types and futures in various places. This includes the `Box::pin(run())`
-    // here, which prevents the large (non-send) main future alone from overflowing the stack.
-    let result = if let Ok(stack_size) = std::env::var(EnvVars::UV_STACK_SIZE) {
-        let stack_size = stack_size.parse().expect("Invalid stack size");
-        let tokio_main = move || {
-            let runtime = tokio::runtime::Builder::new_current_thread()
-                .enable_all()
-                .thread_stack_size(stack_size)
-                .build()
-                .expect("Failed building the Runtime");
-            // Box the large main future to avoid stack overflows.
-            let result = runtime.block_on(Box::pin(run(cli)));
-            // Avoid waiting for pending tasks to complete.
-            //
-            // The resolver may have kicked off HTTP requests during resolution that
-            // turned out to be unnecessary. Waiting for those to complete can cause
-            // the CLI to hang before exiting.
-            runtime.shutdown_background();
-            result
-        };
-        std::thread::Builder::new()
-            .stack_size(stack_size)
-            .spawn(tokio_main)
-            .expect("Tokio executor failed, was there a panic?")
-            .join()
-            .expect("Tokio executor failed, was there a panic?")
-    } else {
+    //
+    // To normalize this we just spawn a new thread called main2 with a size we can set
+    // ourselves. 2MB is typically too small (especially for our debug builds), while 4MB
+    // seems fine.
```

Comment:
How much does this change the overhead of some thing like `uv run echo "hi"` or `uv run python -c "print(\"hello world\")"`?

Reply:
looking now on windows, but it'll be platform-specific to say the least (could test linux in WSL2?)